src/smallmap_gui.cpp

```diff
@@ -1004,6 +1009,45 @@ class SmallMapWindow : public Window {
 		}
 	}
 
+	/**
+	 * Adds industry names to the smallmap.
+	 * @param dpi the part of the smallmap to be drawn into
+	 */
+	void DrawIndustryNames(const DrawPixelInfo *dpi) const
+	{
+		if (this->map_type != SMT_INDUSTRY) return;
+
+		for (const Industry *i : Industry::Iterate()) {
+			const LegendAndColour &tbl = _legend_from_industries[_industry_to_list_pos[i->type]];
+			if (!tbl.show_on_map) continue;
+
+			const bool is_blinking = i->type == _smallmap_industry_highlight && !_smallmap_industry_highlight_state;
+			if (is_blinking) continue;
+
+			if (_industry_to_name_string_width[i->type] == 0) {
+				_industry_to_name_string_width[i->type] = GetStringBoundingBox(tbl.legend, FS_SMALL).width;
+			}
+			const uint8_t &legend_text_width = _industry_to_name_string_width[i->type];
+			const int vertical_padding = ScaleGUITrad(4);
```

Comment:
Why not use the same vertical positioning as town names?

Reply:
Town names use zero padding and it looks OK since we don't really need to see the exact roads/boundaries of a town. For industries, zero padding feels a bit too crowded to make out the colour of the blobs:

# Industry names
![screenshot of industry names with zero padding](https://github.com/OpenTTD/OpenTTD/assets/1967998/334f99d0-77e2-4307-8c9c-9c21c31c4014)

# Industry and town names
![screenshot of industry and town names with zero padding](https://github.com/OpenTTD/OpenTTD/assets/1967998/8c55168b-4440-4095-b850-b27b5921741f)

So I discretionarily picked `4`, which looks good on my screen (and the preview too imo).

Let me know if there's a design system/philosophy for padding.

Alternatively, we could match paddings for both town and industry names for consistency. Here's how it looks if we use `2` for both:

# Town names
<img width="1064" alt="screenshot of town names with 2 padding" src="https://github.com/OpenTTD/OpenTTD/assets/1967998/2884e9d9-9ce6-48bd-96f8-1a14bb220178">

# Industry names
<img width="1064" alt="screenshot of industry names with 2 padding" src="https://github.com/OpenTTD/OpenTTD/assets/1967998/f751b8e0-00fd-43e6-bd65-f7d8e0759b03">

# Industry and town names
<img width="1064" alt="screenshot of industry and town names with zero padding" src="https://github.com/OpenTTD/OpenTTD/assets/1967998/a8ad518f-ec9b-44b2-bd39-129c12f36f18">